ci/long_running_distributed_tests/workloads/cifar_pytorch_pbt.py

```diff
@@ -0,0 +1,134 @@
+        self._update_avail_resources()
+        # With 10% probability inject failure to a worker.
+        if random.random() < 0.1:
+            # With 10% probability fully terminate the node.
+            should_terminate = random.random() < 0.1
+            autoscaler.commands.kill_node(
+                "/home/ubuntu/ray_bootstrap_config.yaml",
+                yes=True,
+                hard=should_terminate,
+                override_cluster_name=None)
+
+
+def initialization_hook():
+    # Need this for avoiding a connection restart issue on AWS.
+    os.environ["NCCL_SOCKET_IFNAME"] = "^docker0,lo"
+    os.environ["NCCL_LL_THRESHOLD"] = "0"
+
+    # set the below if needed
+    # print("NCCL DEBUG SET")
+    # os.environ["NCCL_DEBUG"] = "INFO"
+
+
+def cifar_creator(config):
+    transform_train = transforms.Compose([
+        transforms.RandomCrop(32, padding=4),
+        transforms.RandomHorizontalFlip(),
+        transforms.ToTensor(),
+        transforms.Normalize((0.4914, 0.4822, 0.4465),
+                             (0.2023, 0.1994, 0.2010)),
+    ])  # meanstd transformation
+
+    transform_test = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Normalize((0.4914, 0.4822, 0.4465),
+                             (0.2023, 0.1994, 0.2010)),
+    ])
+    train_dataset = CIFAR10(
+        root="~/data", train=True, download=True, transform=transform_train)
+    validation_dataset = CIFAR10(
+        root="~/data", train=False, download=False, transform=transform_test)
+
+    if config.get("test_mode"):
+        train_dataset = Subset(train_dataset, list(range(64)))
+        validation_dataset = Subset(validation_dataset, list(range(64)))
+
+    train_loader = DataLoader(
+        train_dataset, batch_size=config[BATCH_SIZE], num_workers=2)
+    validation_loader = DataLoader(
+        validation_dataset, batch_size=config[BATCH_SIZE], num_workers=2)
+    return train_loader, validation_loader
+
+
+def optimizer_creator(model, config):
+    """Returns optimizer"""
+    return torch.optim.SGD(
+        model.parameters(),
+        lr=config.get("lr", 0.1),
+        momentum=config.get("momentum", 0.9))
+
+
+ray.init(address="auto", log_to_driver=True)
+num_training_workers = 3
+
+executor = FailureInjectorExecutor(queue_trials=True)
+
+TorchTrainable = TorchTrainer.as_trainable(
+    model_creator=ResNet18,
+    data_creator=cifar_creator,
+    optimizer_creator=optimizer_creator,
+    loss_creator=nn.CrossEntropyLoss,
+    initialization_hook=initialization_hook,
+    num_workers=num_training_workers,
+    config={
+        BATCH_SIZE: 128 * num_training_workers,
+    },
+    use_gpu=True)
+
+pbt_scheduler = PopulationBasedTraining(
+    time_attr="training_iteration",
+    metric="val_loss",
+    mode="min",
+    perturbation_interval=1,
+    hyperparam_mutations={
+        # distribution for resampling
+        "lr": lambda: np.random.uniform(0.001, 1),
```

Comment:
we should probably implement a dry-run too, just to run it on our CI and make sure abstractions don't break

Reply:
@richardliaw what do you mean by this?